### PR TITLE
Add ability to customize OTA responses in DarwinFWTool

### LINKED ISSUE #22560: [Platform] Add ability to customise OTA responses in DarwinFWTool
### Reproduction steps

```bash
Missing functionality.

This issue is manually created for #22277
```

### Platform

darwin

### Platform Version(s)

_No response_

### Type

Manually tested with SDK

### (Optional) If manually tested please explain why this is only manually tested

_No response_

### Anything else?

Manually created for #22277

### PR DESCRIPTION
**Pull request by krypton36**
#### Problem
* Adds ability to set OTA reply parameters for testing Matter.framework

#### Change overview
* Removes previous command to set user consent
* Adds new command to set multiple reply params in OTA
* Adds ability to set action, status, delayed action time, timed invoke timeout, consent, and user consent needed.

New command to set params in interactive mode:
```cli
otasoftwareupdateapp set-reply-params [--commissioner-name] [--action] [--status] [--consent] [--consentNeeded] [--delayedActionTime] [--timedInvokeTimeoutMs]
```

Fixes #22560

#### Testing
* @liamgonyea was able to test the various scenarios in setting the params.

### CODE REVIEW

**Review by liamgonyea — APPROVED** [review]
Changes look good to me.

I tested these against the OTA-Requestor app on sve-2 branch commit: 5471b35f09beafd7db4a5e6ccb9ca2e63d7c9010

**Comment by andy31415** [comment]
Accepted for 1.0: changes in /examples subdirectory only, also seems platform-specific (darwin)

File: examples/darwin-framework-tool/commands/provider/OTAProviderDelegate.mm [hunk 1]
```diff
@@ -72,35 +77,13 @@ - (void)handleQueryImageForNodeID:(NSNumber * _Nonnull)nodeID
...
 
     _selectedCandidate.updateToken = [self generateUpdateToken];
+    NSLog(@"Query Image Status: %hhu", _queryImageStatus);
+    _selectedCandidate.status = @(_queryImageStatus);
 
-    if (params.requestorCanConsent.integerValue == 1) {
-        _selectedCandidate.status = @(MTROtaSoftwareUpdateProviderOTAQueryStatusUpdateAvailable);
-        _selectedCandidate.userConsentNeeded
-            = (_userConsentState == OTAProviderUserUnknown || _userConsentState == OTAProviderUserDenied) ? @(1) : @(0);
+    if (_userConsentNeeded) {
```
Comment (bzbarsky-apple) [inline examples/darwin-framework-tool/commands/provider/OTAProviderDelegate.mm:83]: Shouldn't this still take `params.requestorCanConsent` into account?

Comment (krypton36) [reply examples/darwin-framework-tool/commands/provider/OTAProviderDelegate.mm]: Yes, indeed

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
@krypton36 - merge conflicts, but ready to merge after that.

**Pull request merged**